src/Avalonia.Controls/Mixins/SelectableMixin.cs

```diff
@@ -42,7 +42,7 @@ public static void Attach<TControl>(AvaloniaProperty<bool> isSelected)
         {
             Contract.Requires<ArgumentNullException>(isSelected != null);
 
-            isSelected.Changed.Subscribe(x =>
+            isSelected.Changed.Subscribe((AvaloniaPropertyChangedEventArgs x) =>
```

Comment:
Would probably best to change these instances to use the new typed args, e.g. use `x.NewValue.GetValueOrDefault<bool>()`.

Reply:
I wanted to keep the PR simple, and I think I don't fully understand `BindingValue` yet.

For example, on property changed handlers I always get a `BindingValue` with `HasValue == true`, is this always the case? Also, for example in this specific case, the value is used as `(bool)x.NewValue`, which would fail if it was `null` I think, so in fact the value is always a `bool`, which leads me to believe that it always has a value, could you confirm if this is true?

Anyway, if you think those changes should be part of this PR, I can do it.

Reply:
Yeah, I tried to explain this in the [API docs](https://github.com/AvaloniaUI/Avalonia/blob/d946e3a40e88b49e345740613fa5cba8d7b26c94/src/Avalonia.Base/AvaloniaPropertyChangedEventArgs%601.cs#L56-L59) but it's maybe not as clear as it could be...

Basically by the time the notification gets to `PropertyChanged`/`AvaloniaProperty.Changed`, it's an effective value change meaning that there will always be a value (which, if the property is unset will be the default property value).

But calling .GetValueOrDefault<bool>()` would handle `HasValue == false` anyway and return `false`.

Reply:
I don't really mind if we make that change part of this PR... Maybe it would be better to change all uses of `AvaloniaPropertyChangedEventArgs` to the generic version in a separate PR.